snapcraft/internal/pluginhandler/_stage_package_handler.py

```diff
@@ -0,0 +1,100 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import logging
+
+from snapcraft.internal import repo
+from snapcraft.internal.pluginhandler import stage_package_grammar
+
+logger = logging.getLogger(__name__)
+
+
+class StagePackageHandler:
+    """Interpret stage-packages grammar, and fetch/unpack stage-packages.
+
+    Basic example:
+    >>> import os
+    >>> import tempfile
+    >>> with tempfile.TemporaryDirectory() as tmp:
+    ...    cache_dir = os.path.join(tmp, 'cache')
+    ...    unpack_dir = os.path.join(tmp, 'unpack')
+    ...    handler = StagePackageHandler(['foo'], cache_dir)
+    ...    handler.fetch()
+    ...    handler.unpack(unpack_dir)
+    """
+
+    def __init__(self, stage_packages_grammar, cache_dir, *, sources=None,
+                 project_options=None):
+        """Create new StagePackageHandler.
+
+        Arguments:
```

Comment:
I asked @elopio about it, he said to leave it until he had done his research since other docstrings would need to change as well.

@elopio is now a good time to change these?